File: demo/csharp/seasons/seasons.cs
using System;
using CosineKitty;
using demo_helper;

namespace seasons
{
    class Program
    {
        static int Main(string[] args)
        {
            int year;
            if (args.Length != 1 || !int.TryParse(args[0], out year))
            {
                Console.WriteLine("ERROR: Must provide a year value on the command line.");
                return 1;
            }
            SeasonsInfo seasons = Astronomy.Seasons(year);
            Console.WriteLine("March equinox     : {0}", seasons.mar_equinox);
            Console.WriteLine("June solstice     : {0}", seasons.jun_solstice);
            Console.WriteLine("September equinox : {0}", seasons.sep_equinox);
            Console.WriteLine("December solstice : {0}", seasons.dec_solstice);
            return 0;
        }
    }
}

File: demo/java/src/main/java/io/github/cosinekitty/astronomy/demo/Seasons.java
package io.github.cosinekitty.astronomy.demo;
import io.github.cosinekitty.astronomy.*;

public class Seasons {
    /**
     * Print the times of the equinoxes and solstices for a calendar year.
     *
     * @param year
     * The calendar year value for which to find equinoxes and solstices.
     */
    public static int run(int year) {
        SeasonsInfo seasons = Astronomy.seasons(year);
        System.out.printf("March equinox     : %s%n", seasons.getMarchEquinox());
        System.out.printf("June solstice     : %s%n", seasons.getJuneSolstice());
        System.out.printf("September equinox : %s%n", seasons.getSeptemberEquinox());
        System.out.printf("December solstice : %s%n", seasons.getDecemberSolstice());
        return 0;
    }
}
